### PR TITLE
Refactor ToC Generation

### PR DESCRIPTION
**Pull request by dbolack-ab**
This restructures the existing ToC generation to provide three options in preparation Widget based activities.
    
This changes the existing menu item to a menu parent with subsnippets for generating ToC that use
     - Headers # to ###
     - Headers # to ####
     - Headers # to #####
    
Current styling is set up to emulate the existing styles but probably should be examined further.

### CODE REVIEW

File: themes/V3/5ePHB/snippets.js [hunk 1]
```diff
@@ -21,9 +21,26 @@ module.exports = [
 		view      : 'text',
 		snippets  : [
 			{
-				name : 'Table of Contents',
-				icon : 'fas fa-book',
-				gen  : TableOfContentsGen
+				name        : 'Table of Contents',
```
Comment (5e-Cleric) [inline themes/V3/5ePHB/snippets.js:24]: we might want to change those icons to something like `h1-h4`

File: themes/V3/5ePHB/snippets/tableOfContents.gen.js [hunk 2]
```diff
@@ -27,7 +27,31 @@ const getTOC = (pages)=>{
...
+		if(!_.last(res)) add1(null, page);
+		if(!_.last(_.last(res).children)) add2(null, page);
+		if(!_.last(_.last(_.last(res).children))) add3(null, page);
+		_.last(_.last(_.last(res).children).children).children.push({
+			title    : title,
+			page     : page + 1,
+			children : []
+		});
+	};
+
```
Comment (5e-Cleric) [inline themes/V3/5ePHB/snippets/tableOfContents.gen.js:40]: this looks very readable, thinking if we can refactor this to make it more clear

**Review by 5e-Cleric — APPROVED** [review]
Overall looks fine, left some non-helpful comments

File: themes/V3/5ePHB/snippets/tableOfContents.gen.js (hunk 2 above)
Comment (dbolack-ab) [reply themes/V3/5ePHB/snippets/tableOfContents.gen.js:40]: What needs clarification?

Comment (5e-Cleric) [reply themes/V3/5ePHB/snippets/tableOfContents.gen.js:40]: I just think code should be readable to a point, and even knowing JS i have no idea what that code does, maybe add a simple comment about it?

File: themes/V3/5ePHB/snippets.js (hunk 1 above)
Comment (5e-Cleric) [reply themes/V3/5ePHB/snippets.js:24]: I was thinking in the lines of this:
image

But those are pro icons, and we don't have access to that. I guess it is fine as it is.

**Comment by ericscheid** [comment]
Now that we've sorted out the bugs from PR #3261 .. could we use just the one method for determining if a heading is included or not to avoid confusing authors?

We should also add a snippet to the Style Editor snippet menu for "Include H4, H5 in ToC".

Something like this
```
/* Include H4 and H5 headings in Table of Contents */
/* This only takes effect when the ToC is generated in the brew editor */
h4, h5 { --TOC: include; }`
```

**Comment by dbolack-ab** [comment]
Closing as sorta redundant.

**Pull request closed without merging**